### PR TITLE
Add connectionStateTtl

### PR DESCRIPTION
**Pull request by funkyboy**
In preparation for https://github.com/ably/ably-java/issues/358 

- [x] Remove `TIMEOUT_SUSPEND`
- [x] add `connectionStateTtl` and use it instead of `TIMEOUT_SUSPEND`

### CODE REVIEW

File: lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java [hunk 1]
```diff
@@ -388,4 +389,32 @@ public void run() {
...
+	 */
+	@Test
+	public void connection_details_has_ttl() {
+		try {
+			ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+			final AblyRealtime ably = new AblyRealtime(opts);
+			ably.connection.on(ConnectionEvent.connected, new ConnectionStateListener() {
+				@Override
+				public void onConnectionStateChanged(ConnectionStateChange state) {
+					assertEquals(1, 1);
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: What's the point of this?

Comment (mattheworiordan) [reply lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: Why is this still here?

File: lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java [hunk 2]
```diff
@@ -388,4 +389,32 @@ public void run() {
...
+					try {
+						Field field = ably.connection.connectionManager.getClass().getDeclaredField("connectionStateTtl");
+						field.setAccessible(true);
+						assertEquals("Verify connectionStateTtl has the default value", field.get(ably.connection.connectionManager), 120000L);
+					} catch (NoSuchFieldException|IllegalAccessException e) {
+						fail("Unexpected exception in checking connectionStateTtl");
+					}
+					ably.close();
+				}
+			});
```
Comment (mattheworiordan) [inline lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: There is no wait as far as I can tell so the `onConnectionStateChanged` is not necessarily waited for?  How do you know this test even run and assertion succeeded?

File: lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java (hunk 1 above)
Comment (funkyboy) [reply lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: @paddybyers @mattheworiordan Removed https://github.com/ably/ably-java/pull/389/commits/26c18f67a27d90d292ad53e3fed881f645bde190

File: lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java (hunk 2 above)
Comment (funkyboy) [reply lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: @mattheworiordan I made the test more explicit https://github.com/ably/ably-java/pull/389/commits/850eaf9d2a7f186bcd73ad1594903142eccd0086

File: lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java [hunk 3]
```diff
@@ -408,9 +408,10 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
 					} catch (NoSuchFieldException|IllegalAccessException e) {
 						fail("Unexpected exception in checking connectionStateTtl");
 					}
-					ably.close();
 				}
 			});
+			new Helpers.ConnectionManagerWaiter(ably.connection.connectionManager).waitFor(ConnectionState.connected);
```
Comment (paddybyers) [inline lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: I agree this is an improvement, but even better would have been to have an explicit check that the callback ran

Comment (funkyboy) [reply lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java]: @paddybyers done https://github.com/ably/ably-java/pull/389/commits/9de807f16a0dc59c228b2701db163f23f752dfc8

**Pull request merged**